### DISCUSSION TITLE
How can I connect TabPy to Tableau using two different machines?

### DISCUSSION THREAD

**Discussion by mitchell-leon** [tableau/TabPy#542] [Q&A]
I have Tableau Desktop on my local machine and Anaconda on a virtual desktop. I have successfully installed TabPy on my virtual desktop but cannot connect to Tableau Desktop as they are on different machines. Does anyone know how I can connect TabPy using a virtual desktop and local machine?

**Comment by aviad-schiffmann** [comment]
i believe you can expose the port you asked tabpy server to use (9004) out from your virtual machine to your desktop.

**Comment by nmannheimer** [comment]
Yes, as @aviad-schiffmann says, you need to ensure TabPy is running in the VM, and ensure that the two machines can connect over your network on the specific port TabPy is listening on, which by default is 9004.
